src/foam/comics/v2/DAOControllerConfig.js

```diff
@@ -137,6 +137,42 @@ foam.CLASS({
           : of.getAxiomsByClass(foam.core.Property).map(p => p.name);
       }
     },
+    {
+      class: 'StringArray',
+      name: 'searchColumns',
+      factory: null,
+      expression: function(of, defaultColumns) {
+        var tableSearchColumns = of.getAxiomByName('searchColumns');
+
+        var filteredDefaultColumns = defaultColumns.filter(c => {
+          //  to account for nested columns like approver.legalName
+          if ( c.split('.').length > 1 ) return false;
+
+          var a = of.getAxiomByName(c);
+
+          if ( ! a ) console.warn("Column does not exist for " + of.name + ": " + c);
+          
+          return a
+            && ! a.storageTransient 
+            && ! a.networkTransient
+            && a.searchView 
+            && ! a.hidden
+        });
+
+        var allProps = of.getAxiomsByClass(foam.core.Property).filter(p => {
+          return ! p.storageTransient 
+            && ! p.networkTransient
+            && p.searchView 
+            && ! p.hidden
+        })
+
+        return tableSearchColumns
+          ? tableSearchColumns.columns
+          : filteredDefaultColumns 
+            ? filteredDefaultColumns
+            : allProps
+      }
+    },
```

Comment:
If the return ternary statement was set as a conditional if statements - you can potentially avoid 2 loops through an array - filteredDefaultColumns and allProps.
 So 
 if ( tableSearchColumns ) return tableSerachColumns.columns;
filteredDefaultColumns = ...
if ( filteredDefaultColumns ) return filteredDefaultColumns;
...

Reply:
Yeah could do that instead of the ternary. Was initially a ternary when there was only 2 conditions. Thought conttinuing it would look cleaner and there isn't much inefficiency seeing as searchColumns would usually never be bigger than like 10. However the all props could end up being from a massive model I guess